src/main.rs

```diff
@@ -88,21 +103,46 @@ fn run() -> Result<()> {
     Ok(())
 }
 
-fn main() {
-    if let Err(e) = &run() {
-        let stderr = io::stderr();
-        let mut stderr = stderr.lock();
+fn setup_logging(matches: &clap::ArgMatches) {
+    // * Log at info by default.
+    // * Allow users the option of setting complex logging filters using
+    //   env_logger's `RUST_LOG` environment variable.
+    // * Override both of those if the logging level is set via the `--log`
+    //   command line argument.
+    let env = env_logger::Env::default()
+        .filter_or(env_logger::DEFAULT_FILTER_ENV, "info");
+    let mut builder = env_logger::Builder::from_env(env);
+    builder.default_format_timestamp(false);
+
+    let log_lvl_from_env =
```

Comment:
This looks shady. Why store it in a variable and then ignore it?

Reply:
I have no idea. All commits except mine are already in master.

This PR is only due to:
https://github.com/rust-embedded/svd2rust/pull/308#issuecomment-511140315